File: Step 03 Arrays/Step 3.1 Easy/Second Largest element/Second Largest element.cpp
//User function template for C++
class Solution{
public:	
	// Function returns the second
	// largest elements
	int print2largest(int arr[], int n) {
	    // code here
	    int first_maxx = -1;
	    int sec_maxx = -1;
	    for (int i=0; i<n; ++i) {
	        int x = arr[i];
	        if (x > first_maxx) {
	            sec_maxx = first_maxx;
	            first_maxx = x;
	        } else if (x < first_maxx && x > sec_maxx)
	            sec_maxx = x;
	    }
	    return sec_maxx;
	}
};

File: Step 03 Arrays/Step 3.1 Easy/Second Largest element/Second Largest element.java


//User function Template for Java

class Solution {
    int print2largest(int arr[], int n) {
        // code here
        int firstMax = -1;
        int secondMax = -1;
    
        for (int i = 0; i < arr.length; i++) {
            int x = arr[i];
            if (x > firstMax) {
                secondMax = firstMax;
                firstMax = x;
            } else if (x < firstMax && x > secondMax) {
                secondMax = x;
            }
        }
        
        return secondMax;
    }
}